kube-kvm/provision-kube-host-kvm.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+# This script is intended to be run on a SLES box installed as a KVM host.
+# After OS installation, the following commands should also be run to set up directory
+# permissions and the default network (as root):
+# - usermod -aG libvirt root
+# - virsh net-define net-default.xml
+# - virsh net-start default
+# - ntpdate time.nist.gov
+# - systemctl enable ntpd
+# - systemctl start ntpd
+
+set -o errexit
+
+export KUBE_VM_NAME=${KUBE_VM_NAME:-$1}
+export KUBE_VM_IMAGE_NAME=${KUBE_VM_IMAGE_NAME:-scf-libvirt-v2.0.7}
+export KUBE_VM_IMAGE_PATH=${KUBE_VM_IMAGE_PATH:-~/qcow2-disks}
+export KUBE_VM_MEM_GIB=${KUBE_VM_MEM_GIB:-8}
+
+mkdir -p "$KUBE_VM_IMAGE_PATH"
+cd "$KUBE_VM_IMAGE_PATH"
+
+if [[ -z "$KUBE_VM_NAME" ]]; then
+  echo "A VM name must be provided"
+  exit 1
+fi
+
+# Fetch base image if it doesn't exist
+if [[ ! -f $KUBE_VM_IMAGE_NAME.qcow2 ]]; then
+  echo "Fetching base image:"
+  curl -L https://cf-opensusefs2.s3.amazonaws.com/vagrant/${KUBE_VM_IMAGE_NAME}.box | tar --wildcards box.img -xz
+  mv box.img $KUBE_VM_IMAGE_NAME.qcow2
+fi
+
+# Fetch VM access key if it doesn't exist
+if [[ ! -f vm-key ]]; then
+  curl -sL -o "vm-key" https://raw.githubusercontent.com/mitchellh/vagrant/v1.9.6/keys/vagrant
+  chmod 400 vm-key
+fi
+KUBE_VM_KEY=${KUBE_VM_KEY:-$PWD/vm-key}
+
+if [[ $KUBE_VM_MEM_GIB -gt $(free -g | awk 'NR == 2 { print $2 - 2 }') ]]; then
+  echo KUBE_VM_MEM_GIB value $KUBE_VM_MEM_GIB exceeds system resources
+  exit 1
+fi
+
+unset RUNNING_VM_IMAGE
+if virsh domstate $KUBE_VM_NAME 2>/dev/null; then
+  echo "Deleting existing domain $KUBE_VM_NAME"
+  RUNNING_VM_IMAGE=$(virsh dumpxml $KUBE_VM_NAME | grep 'source file' | grep -o "'.*'" | tr -d "'")
+  if [[ -z $RUNNING_VM_IMAGE ]]; then
```

Comment:
Please quote things that might be empty.  Ideally quote everything, but especially things that might be empty.